middleware/middleware.js

```diff
@@ -60,6 +61,50 @@ module.exports.fetchRandomGameData = async (req, res, next) => {
   next();
 };
 
+//function for making API requests for additional data
+module.exports.fetchAdditionalGameData = async (fetchLink) => {
+  let response = null;
+  try {
+    response = await axios.get(fetchLink);
+  } catch (error) {
+    if (process.env.IN_DEVELOPMENT === "true") {
+      console.log(error.response.status);
+    }
+  }
+  return response.data;
+};
+
+//adds more detailed game data (stores, game website and developers) and stores in database
+module.exports.completeAndStoreGameData = async (detailedGameData) => {
```

Comment:
I would rewrite this segment like this to make it a bit more structured and easier to read:

```suggestion
module.exports.completeAndStoreGameData = async (detailedGameData) => {
  //making API requests to fetch developers, website and store links for the game
  const GAME_DATA_URL = `${apiUrl}games/${detailedGameData.gameId}?key=${process.env.RAWG_KEY}`;
  const STORES_URL = `${apiUrl}games/${detailedGameData.gameId}/stores?key=${process.env.RAWG_KEY}`;

  const gameData = await this.fetchAdditionalGameData(GAME_DATA_URL);
  const gameStoresResponse = await this.fetchAdditionalGameData(STORES_URL);

  const developerNames = gameData.developers.map((developer) => developer.name);
  const gameStoresLinks = gameStoresResponse.results.map(
    ({ url, store_id }) => ({ url, store_id })
  );

  const updatedGameData = {
    ...detailedGameData,
    website: gameData.website,
    developers: developerNames,
    stores: gameStoresLinks,
  };

  //saving data in mongodb
  const newGameData = new GameData({
    ...updatedGameData,
  });
  await newGameData.save();
};

```